fix: build rechazó: Build falla en `:app:composeApp:bundleBusinessReleaseClassesToRun

## Contexto\n\nDetectado al analizar el rechazo del issue #2023 (**qa(verificación): Endpoints backend core — auth, productos, pedidos, delivery**).\n\n| Campo | Valor |\n|-------|-------|\n| **Issue bloqueado** | #2023 |\n| **Agente que falló** | build |\n| **Fase** | build |\n| **Prioridad** | 🔴 Alta — bloqueo directo |\n| **Fuente** | agent-verdict |\n\n## Causa raíz\n\n### build rechazó: Build falla en `:app:composeApp:bundleBusinessReleaseClassesToRuntimeJar` por dependencia de tarea no satisfecha.\n\nBuild falla en `:app:composeApp:bundleBusinessReleaseClassesToRuntimeJar` por dependencia de tarea no satisfecha. Excluir `compileBusinessReleaseKotlinAndroid` (exclusion obligatoria para issues sin label `app:*`) deja a la tarea `bundleBusinessReleaseClassesToRuntimeJar` intentando leer el output del compile excluido.\n\nError (de `.pipeline/logs/build-2023.log`, lineas 485-495):\n\n  > Task :app:composeApp:bundleBusinessReleaseClassesToRuntimeJar FAILED\n\n  FAILURE: Build failed with an exception.\n\n  * What went wrong:\n  Execution failed for task ':app:composeApp:bundleBusinessReleaseClassesToRuntimeJar'.\n  > Querying the mapped value of provider(java.util.Set) before task ':app:composeApp:compileBusinessReleaseKotlinAndroid' has completed is not supported\n\n  BUILD FAILED in 1m 6s\n  190 actionable tasks: 13 executed, 9 from cache, 168 up-to-date\n\nContexto:\n- Branch `agent/2023-backend-dev` fast-forward de `origin/main` (27 commits traidos, merge limpio).\n- Issue #2023 tiene labels `area:backend`, `area:testing`, `feature:checkout-pagos`, `priority:critical`, sin labels `app:*`, por lo que se corrio `./gradlew check --no-daemon` con las exclusiones de Release Android indicadas en el rol.\n- La branch no tiene commits propios todavia; la falla es reproducible sobre main sin cambios del dev.\n\nAccion sugerida para el developer/pipeline: agregar `-x bundleBusinessReleaseClassesToRuntimeJar` (y las variantes Client/Delivery y `bundle*ReleaseClassesToCompileJar`) junto con las exclusiones `compile*ReleaseKotlinAndroid`, o que el wiring de Gradle no evalue los bundles cuando el compile esta excluido. El bug es de configuracion de build, no de logica de negocio.\n\n## Criterios de aceptación\n\n- [ ] El problema descrito está corregido\n- [ ] El issue #2023 puede reintentarse sin este bloqueo\n- [ ] No introduce regresiones\n\n---\n_Issue creado automáticamente por el rejection report v6 (estricto, 1 causa por reporte)._

## Criterios de aceptación (PO)

Basados en el análisis técnico (Guru) y la auditoría de seguridad (Security). Este issue es un fix de configuración del pipeline — los criterios son todos verificables por lectura de archivo + ejecución local del comando.

### Alcance del cambio

- Archivo objetivo: `.pipeline/roles/build.md` (Opción A del Guru — ampliación de exclusiones).
- Opcionalmente `app/composeApp/build.gradle.kts` si se elige Opción B (property Gradle). Si se va por Opción A, dejar Opción B registrada como follow-up.

### Criterios funcionales

1. **Dado** un issue SIN labels `app:*` entrando a la fase de build, **cuando** el agente `/builder` ejecuta el comando Gradle definido en `.pipeline/roles/build.md`, **entonces** el comando incluye exclusiones explícitas para las 3 dimensiones (`client`, `business`, `delivery`) de:
   - `bundle{Flavor}ReleaseClassesToRuntimeJar`
   - `bundle{Flavor}ReleaseClassesToCompileJar`
   - `lintVital{Flavor}Release`
   - `lint{Flavor}Release`
   además de las `compile{Flavor}ReleaseKotlinAndroid` ya existentes.

2. **Dado** el issue #2023 (u otro sin `app:*`) reencolado a build, **cuando** corre el gate, **entonces** el log NO muestra el error `"Querying the mapped value of provider(java.util.Set) before task ':app:composeApp:bundle*ReleaseClassesToRuntimeJar' has completed is not supported"` y la tarea `BUILD FAILED` ya no aparece por esa causa.

3. **Dado** un issue con algún label `app:client`, `app:business` o `app:delivery`, **cuando** entra a la fase de build, **entonces** las exclusiones Android Release NO se aplican (el build ejecuta completo, incluyendo bundle/lint Release del flavor correspondiente). Esto debe quedar explícito en el rol como condicional por presencia de `app:*`.

### Criterios de no-regresión

4. **Dado** el workflow de GitHub Actions en push a `main`, **cuando** corre el build completo, **entonces** NO está afectado por las nuevas exclusiones (el rol de pipeline por-issue no modifica `.github/workflows/**`).

5. **Dado** el comando resultante, **cuando** se inspecciona, **entonces** NO excluye:
   - `:backend:check` ni `:users:check` (gates backend críticos).
   - `verifyNoLegacyStrings` (gate de compliance de strings).
   - Ninguna tarea `signingConfig*`, `sign*`, ni nada que deshabilite firma de APK.

### Criterios de documentación

6. **Dado** el rol `.pipeline/roles/build.md`, **cuando** un developer lo lee, **entonces** encuentra una sección que:
   - Enumera las exclusiones Android Release y su justificación (bug Gradle+AGP, referencias a gradle/gradle#21290 y b/244354876).
   - Aclara explícitamente que las exclusiones aplican **sólo al gate por-issue sin labels `app:*`**.
   - Aclara que el CI de GitHub Actions en push a `main` ejecuta el build completo sin estas exclusiones como gate final.

### Criterios de verificación local (antes de aprobar)

7. **Dado** el fix aplicado, **cuando** el dev/build ejecuta el comando Gradle reproducido del rol sobre la rama correspondiente (`agent/2393-*` o main), **entonces** el comando finaliza con `BUILD SUCCESSFUL` (o con fallos de otros tests, pero NO con el error de `bundle*ReleaseClassesToRuntimeJar`).

### Out of scope (no forman parte de este issue)

- El bug de compilación real en `compileBusinessReleaseKotlinAndroid` (líneas 647-737 del log #2023) que Guru detectó como colateral — se abre issue aparte si se confirma en main.
- Migración a Opción B (`-Pintrale.skipAndroidRelease=true`) — queda como follow-up si se aprueba el refactor.
- Refactor a targets acotados (Opción C) — queda como follow-up.

### Labels sugeridos para el issue antes de dev

- Mantener: `priority:high`, `qa:dependency`.
- Agregar: `area:infra`, `tipo:infra` (es cambio de tooling del pipeline, sin impacto en producto de usuario).
- Agregar: `qa:skipped` con justificación: "Cambio en `.pipeline/roles/build.md` sin impacto en UI ni endpoints; verificación por ejecución local del build del issue bloqueado #2023". Esto permite al PO de la fase de aprobación saltar el video según el PASO 0.A del rol.

### Valor entregado al usuario

Ninguno directo — es deuda técnica de infraestructura. El valor es **desbloquear la cola de issues backend** (empezando por #2023 "Endpoints backend core — auth, productos, pedidos, delivery"), que sí entregan valor al usuario final.

--- a/.pipeline/roles/build.md
+++ b/.pipeline/roles/build.md
@@ -57,7 +57,13 @@ Si el issue **NO tiene ningún label `app:*`** (`HAS_APP_LABEL=0`), corré con e
   -x compileDeliveryReleaseUnitTestKotlinAndroid \
   -x testClientReleaseUnitTest \
   -x testBusinessReleaseUnitTest \
-  -x testDeliveryReleaseUnitTest
+  -x testDeliveryReleaseUnitTest \
+  -x lintVitalClientRelease \
+  -x lintVitalBusinessRelease \
+  -x lintVitalDeliveryRelease \
+  -x lintClientRelease \
+  -x lintBusinessRelease \
+  -x lintDeliveryRelease
 ```
 
 Si el issue **tiene algún label `app:*`** (`HAS_APP_LABEL>0`), corré sin exclusiones de Release Android (se necesitan para el APK del paso 4):
@@ -70,15 +76,20 @@ Si el issue **tiene algún label `app:*`** (`HAS_APP_LABEL>0`), corré sin exclu
   -x compileTestKotlinIosSimulatorArm64
 ```
 
-Las exclusiones permanentes (`WasmJs` test y iOS) son siempre obligatorias. Las de `compile*ReleaseKotlinAndroid` (y toda la cadena Release Android Unit Test) se aplican solo cuando no hay cambios que afecten a Android, para evitar OOM y tiempos de build largos en cambios de backend puros.
+Las exclusiones permanentes (`WasmJs` test y iOS) son siempre obligatorias. Las de `compile*ReleaseKotlinAndroid` (y toda la cadena Release Android Unit Test + lint) se aplican solo cuando no hay cambios que afecten a Android, para evitar OOM y tiempos de build largos en cambios de backend puros.
+
+> **Alcance de las exclusiones Android Release.** Las exclusiones `compile*Release*`, `bundle*Release*`, `ksp*Release*`, `test*ReleaseUnitTest`, `lint*Release` y `lintVital*Release` se aplican **únicamente al gate por-issue del pipeline cuando el issue no tiene label `app:*`**. El workflow de GitHub Actions en push a `main` (`.github/workflows/**`) no usa este rol — corre el build completo sin estas exclusiones como gate final antes de deploy a Lambda.
 
 > Importante — cadena de dependencias Release: cuando excluís `compile<Flavor>ReleaseKotlinAndroid` para ahorrar RAM, Gradle 8.13 no permite query del provider mapeado antes de que la tarea productora complete. Eso rompe la cadena completa de tareas que leen esos outputs:
 > 1. `bundle<Flavor>ReleaseClassesToRuntimeJar` / `ToCompileJar` (consumen el .jar del compile)
 > 2. `ksp<Flavor>ReleaseUnitTestKotlinAndroid` (transform del classes.jar del bundle ToCompileJar)
 > 3. `compile<Flavor>ReleaseUnitTestKotlinAndroid` (depende del KSP)
 > 4. `test<Flavor>ReleaseUnitTest` (depende del compile unit test)
+> 5. `lint<Flavor>Release` / `lintVital<Flavor>Release` (consumen el jar del bundle via `AndroidLintAnalysisTask`)
 >
-> Por eso hay que excluir TODA la cadena Release UnitTest, no solo el compile raíz. El síntoma típico es `Querying the mapped value of provider(java.util.Set) before task [...] has completed is not supported` o `Failed to transform classes.jar`.
+> Por eso hay que excluir TODA la cadena Release (compile + bundle + ksp unit test + compile unit test + test unit test + lint + lintVital), no solo el compile raíz. El síntoma típico es `Querying the mapped value of provider(java.util.Set) before task [...] has completed is not supported` o `Failed to transform classes.jar`.
+>
+> Referencias del bug upstream: [gradle/gradle#21290](https://github.com/gradle/gradle/issues/21290) (Gradle) y [b/244354876](https://issuetracker.google.com/issues/244354876) (Android Gradle Plugin).
 
 Si el build falla, saltar directamente al paso 5 (reportar rechazado).
 

